lib/collection/src/common/snapshots_manager.rs

```diff
@@ -60,8 +61,13 @@ impl SnapshotStorageManager {
                 Ok(SnapshotStorageManager::LocalFS(SnapshotStorageLocalFS))
             }
             SnapshotsStorageConfig::S3 => {
-                let mut builder = AmazonS3Builder::new();
-                if let Some(s3_config) = &snapshots_config.s3_config {
+                let builder = if let Some(s3_config) = &snapshots_config.s3_config {
```

Comment:
Possibly could just replace AmazonS3Builder::new with AmazonS3Builder::from_env method, but not sure if there are cases where this would not be desirable behavior.

Reply:
I cannot think of a case where we would not want to source details from the environment. Let's remove `with_env` from our config, and always use [`from_env`](https://docs.rs/object_store/latest/object_store/aws/struct.AmazonS3Builder.html#method.from_env).

If this ever becomes an issue in the future, we can always add the config option back.